tools/bridge/bridge/validation_utils.py

```diff
@@ -0,0 +1,73 @@
+from web3 import Web3
+from web3.contract import Contract
+from web3._utils.abi import abi_to_signature
+
+from bridge.contract_abis import MINIMAL_VALIDATOR_PROXY_ABI
+
+
+def validate_contract(contract: Contract) -> None:
+    """Verifies if a given contract exists on the chain.
+
+    It checks if any code is stored for the address. For events and functions
+    in the ABI it will be checked if their signature hash can be found within
+    the code. Throws an exception if the contract could not be verified. Else
+    return nothing.
+    """
+
+    contract_code = contract.web3.eth.getCode(contract.address)
+
+    if not contract_code:
+        raise ValueError(
+            f"The given contract address {contract.address} does not point to a contract!"
+        )
+
+    for description in contract.abi:
+        description_type = description.get("type", "function")
+
+        if description_type in ("constructor", "fallback"):
+            continue
+
+        signature = abi_to_signature(description)
+        signature_hash = Web3.keccak(text=signature)
+
+        if description_type == "function":
+            signature_hash = signature_hash[:4]
+
+        if signature_hash not in contract_code:
+            raise ValueError(
+                f"The contract at the given address {contract.address} does not"
+                f" support the {description_type} with the following signature: {signature}!"
+            )
+
+
+def validate_confirmation_permissions(home_bridge_contract: Contract, address: str):
+    validator_proxy_address = home_bridge_contract.functions.validatorProxy().call()
+    validator_proxy_contract = home_bridge_contract.web3.eth.contract(
+        address=validator_proxy_address, abi=MINIMAL_VALIDATOR_PROXY_ABI
+    )
+
+    try:
+        validate_contract(validator_proxy_contract)
```

Comment:
This check seems unrelated to the access permissions. Maybe just remove it and assume that you got a valid contract?

Reply:
It is not. Else I would not know what to respond to the user. If my call fails I would like to give the user a meaningful message if I can. Make a call to the correct contract is something we can check. This could at least help ourselves to verify our bridge setup after deployment.

Reply:
I think about removing or exporting this due to the [decision above ](https://github.com/trustlines-protocol/blockchain/pull/268#discussion_r308181643). We don't want to check this over and over again.

Reply:
I exported this check. It is now done as for the other contracts in the `main` module as well. Therefore a `get_validator_proxy_contract` function has been added, to preserve the functionality of deriving this contract from the home bridge contract. I don't wanted to initialize the confirmation sender with an additional contract. Not sure if you like this. Please add a review therefore.